client/src/app/components/Autocomplete.tsx

```diff
@@ -256,42 +323,41 @@ export const Autocomplete: React.FC<IAutocompleteProps> = ({
     </div>
   );
 
+  const renderMenuItems = () => {
+    const effectiveGroupedOptions = isGrouped
+      ? groupedOptions
+      : { undefined: filteredOptions };
+
+    return Object.entries(effectiveGroupedOptions).map(
+      ([groupName, groupOptions], index) => (
+        <React.Fragment key={groupName || `ungrouped-${index}`}>
+          <MenuGroup label={groupName}>
```

Comment:
1. assuming that the "groupedOptions" contain the "Ungrouped" under empty string we can just ` label={groupName || undefined}`
2. we should actually first try the group label and fallback to the group name